lib/query_components/filter.rb

```diff
@@ -1,23 +1,27 @@
+require "query_components/user_filter"
+require "query_components/visibility_filter"
+require 'elasticsearch/queries'
+
 module QueryComponents
   class Filter < BaseComponent
+    include Elasticsearch::Queries
+
     def payload(excluding = [])
-      rejects = []
-      filters = []
+      user_filter = QueryComponents::UserFilter.new(search_params)
+      visibility_filter = QueryComponents::VisibilityFilter.new(search_params)
 
-      param_filters = search_params.filters.reject do |filter|
-        excluding.include?(filter.field_name)
-      end
+      filters = combine_filters(user_filter.filters_payload(excluding), :and)
```

Comment:
This is a little confusing; I like that you're trying to keep it generic.  I assume that you're not including "visibility_filter" in the combine because you know it will always be empty: however, it would be even nicer to just make and array of "filter_components", and call .map on that to get the input for the combine_filters calls.

Something like:
```ruby
filter_components = [
QueryComponents::UserFilter.new(search_params),
QueryComponents::VisibilityFilter.new(search_params),
]
filters = combine_filters(filter_components.map { |component| component.filters_payload(excluding) }, :and)
```